### PR TITLE
feat(DIA-134): add article slideshow

### PR DESCRIPTION
**Pull request by araujobarret**
This PR resolves [DIA-134] 

### Description

This PR adds the Slideshow route `article/:articleID/slideshow`. It's hidden under a feature flag but we might release it before the next release, we'll host some QA internally on Diamond.

We had no designs for Eigen but we based on the mobile view from Force to replicate it.

iPhone

https://github.com/artsy/eigen/assets/15792853/c20b8c00-98d4-4762-bb73-32676ad8142d

iPad

https://github.com/artsy/eigen/assets/15792853/3dc1ae9b-76bc-467f-b620-47fa05127f83

Android

https://github.com/artsy/eigen/assets/15792853/a41e6083-2edb-47df-9347-dd2a1cb98783

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- feat: add article slideshow route and section elements - araujobarret

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[DIA-134]: https://artsyproduct.atlassian.net/browse/DIA-134?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (feat: add article slideshow route and section elements - araujobarret)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 9b334607f07739ab21333d670dd2a66ef124cf4c
</p>

**Review by gkartalis — COMMENTED** [review]
Question: I don't see the feature flag being used anywhere other than the features.ts file.

Which is the surface that we want hidden behind the feature flag? 

Also on another note should we maybe also use `@skip` from [relay](https://relay.dev/docs/glossary/#skip) to conditionally fetch the data depending on the feature flag status.

File: src/app/Scenes/ArticleSlideShow/Components/ArticleSlideShow.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+          <CloseIcon height={24} width={24} />
+        </Touchable>
+      </Flex>
+
+      <Flex flexDirection="row" flex={1}>
+        <FlashList
+          accessibilityLabel="Image and description rail"
+          data={figures}
+          horizontal
+          keyExtractor={(_, index) => `ArticleSlideShowItem-${index}`}
```
[the remark this answers is not in the document]
Comment (araujobarret) [reply src/app/Scenes/ArticleSlideShow/Components/ArticleSlideShow.tsx]: done in https://github.com/artsy/eigen/pull/9474/commits/2953158d142e450472fb2ff598aa526740dab364

File: src/app/Scenes/ArticleSlideShow/Components/ArticleSlideShowCaption.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+              <Text variant="sm-display" color="black60" italic>
+                {data.title}
+              </Text>
+              {!!data.date && `, ${data.date}`}
+            </Text>
+          )}
+
+          {!!data.partner?.name && (
+            <Text variant="xs" color="black60">
+              {data.partner.name ?? ""}
```
[the remark this answers is not in the document]
Comment (araujobarret) [reply src/app/Scenes/ArticleSlideShow/Components/ArticleSlideShowCaption.tsx]: done in https://github.com/artsy/eigen/pull/9474/commits/2953158d142e450472fb2ff598aa526740dab364

File: src/app/Scenes/Article/Components/Sections/ArticleSectionImageSet.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    return null
+  }
+
+  const cover = section.cover
+
+  return (
+    <Flex p={2}>
+      <Image
+        src={cover.image?.url!}
+        aspectRatio={1.56}
```
[the remark this answers is not in the document]
Comment (araujobarret) [reply src/app/Scenes/Article/Components/Sections/ArticleSectionImageSet.tsx]: To keep the same ratio as Force, done in https://github.com/artsy/eigen/pull/9474/commits/2953158d142e450472fb2ff598aa526740dab364

**Review by damassi — APPROVED** [review]
👌

**Comment by MounirDhahri** [comment]
Hey @araujobarret, this is not released yet although it seems like done. Are you intending on releasing it or is this work on hold?

**Comment by pam-** [comment]
hi @MounirDhahri! We'd like to release it

**Comment by araujobarret** [comment]
We're going to release the slideshow this week, thanks for flagging!

**Comment by MounirDhahri** [comment]
great! you are welcone

**Pull request merged**